src/Tizen.Uix.Stt/Tizen.Uix.Stt/SttClient.cs

```diff
@@ -504,88 +567,132 @@ public SttClient()
         /// Event to be invoked when the default language changes.
         /// </summary>
         /// <since_tizen> 3 </since_tizen>
+        /// <privilege>http://tizen.org/privilege/recorder</privilege>
+        /// <exception cref="InvalidOperationException">This exception can be due to an invalid state.</exception>
+        /// <exception cref="NotSupportedException">This exception can be due to STT not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">This exception can be due to permission denied.</exception>
+        /// <exception cref="ArgumentException">This exception can be due to improper value provided while setting the value.</exception>
+        /// <pre>
+        /// The state must be created.
+        /// </pre>
         public event EventHandler<DefaultLanguageChangedEventArgs> DefaultLanguageChanged
         {
             add
             {
                 lock (thisLock)
                 {
-                    _languageDelegate = (IntPtr handle, IntPtr previousLanguage, IntPtr currentLanguage, IntPtr userData) =>
-                    {
-                        string previousLanguageString = Marshal.PtrToStringAnsi(previousLanguage);
-                        string currentLanguageString = Marshal.PtrToStringAnsi(currentLanguage);
-                        DefaultLanguageChangedEventArgs args = new DefaultLanguageChangedEventArgs(previousLanguageString, currentLanguageString);
-                        _defaultLanguageChanged?.Invoke(this, args);
-                    };
-                    SttError error = SttSetDefaultLanguageChangedCB(_handle, _languageDelegate, IntPtr.Zero);
-                    if (error != SttError.None)
+                    if (State.Created != CurrentState)
                     {
-                        Log.Error(LogTag, "Add DefaultLanguageChanged Failed with error " + error);
+                        Log.Error(LogTag, "Add DefaultLanguageChanged Failed with error " + SttError.InvalidState);
+                        throw ExceptionFactory.CreateException(SttError.InvalidState);
                     }
-
-                    else
+                    if (null == _defaultLanguageChanged)
                     {
-                        _defaultLanguageChanged += value;
+                        _languageDelegate = (IntPtr handle, IntPtr previousLanguage, IntPtr currentLanguage, IntPtr userData) =>
```

Comment:
If change the type of parameter from IntPtr to String, manual marshaling may be not need.